3b.run_sam/scripts/1.run_sam2_microscopy.py

```diff
@@ -0,0 +1,615 @@
+tiff_file_names = [file.stem for file in tiff_files]
+# files to df
+tiff_df = pd.DataFrame({"file_name": tiff_file_names, "file_path": tiff_files})
+
+# split the file_path column by _ but keep the original column
+tiff_df["file_name"] = tiff_df["file_name"].astype(str)
+tiff_df[["Well", "FOV", "Timepoint", "Z-slice", "Channel", "illum"]] = tiff_df[
+    "file_name"
+].str.split("_", expand=True)
+tiff_df["Well_FOV"] = tiff_df["Well"] + "_" + tiff_df["FOV"]
+# drop all channels except for the first one
+tiff_df = tiff_df[tiff_df["Channel"] == "C01"]
+tiff_df = tiff_df.drop(columns=["Channel", "illum"])
+tiff_df["new_path"] = (
+    str(ordered_tiffs)
+    + "/"
+    + tiff_df["Well_FOV"]
+    + "/"
+    + tiff_df["file_name"]
+    + ".tiff"
+)
+tiff_df.reset_index(drop=True, inplace=True)
+tiff_df.head()
+
+
+# In[ ]:
+
+
+# copy the files to the new directory
+# from file path to new path
+for index, row in tiff_df.iterrows():
+    new_path = pathlib.Path(row["new_path"])
+    new_path.parent.mkdir(parents=True, exist_ok=True)
+    shutil.copy(row["file_path"], new_path)
+
+
+# In[ ]:
+
+
+# get the list of directories in the ordered tiffs directory
+ordered_tiff_dirs = list(ordered_tiffs.glob("*"))
+ordered_tiff_dir_names = [dir for dir in ordered_tiff_dirs]
+ordered_tiff_dir_names
+for dir in ordered_tiff_dir_names:
+    out_dir = converted_to_video_dir / dir.name
+    out_dir.mkdir(parents=True, exist_ok=True)
+    for tiff_file in dir.glob("*.tiff"):
+        jpeg_file = pathlib.Path(f"{out_dir}/{tiff_file.stem}.jpeg")
+
+        if not jpeg_file.exists():
+            try:
+                with Image.open(tiff_file) as img:
+                    # Convert the image to 8-bit per channel
+                    img = img.convert("L")
+                    img.save(jpeg_file)
+            except Exception as e:
+                print(f"Failed to convert {tiff_file}: {e}")
+
+
+# In[ ]:
+
+
+# get list of dirs in the converted to video dir
+converted_dirs = list(converted_to_video_dir.glob("*"))
+converted_dir_names = [dir for dir in converted_dirs]
+for dir in converted_dir_names:
+    dir = sorted(dir.glob("*.jpeg"))
+    for i in enumerate(dir):
+        # rename the files to be in order
+        i[1].rename(f"{dir[0].parent}/{str(i[0] + 1).zfill(3)}.jpeg")
+
+
+# ### Donwsample each frame to fit the images on the GPU - overwrite the copies JPEGs
+
+# In[ ]:
+
+
+# get files in the directory
+converted_dirs_list = list(converted_to_video_dir.rglob("*"))
+converted_dirs_list = [f for f in converted_dirs_list if f.is_file()]
+# posix path to string
+files = [str(f) for f in converted_dirs_list]
+
+
+# In[ ]:
+
+
+downscale_factor = 12
+# sort the files by name
+# downsample the image
+for f in files:
+    img = io.imread(f)
+    # downsample the image
+    downsampled_img = img[::downscale_factor, ::downscale_factor]
+    # save the downsampled image in place of the original image
+    io.imsave(f, downsampled_img)
```

Comment:
This seems sus, but is could be me misinterpreting the code comment. Just to confirm, you are not overwriting the original TIFF images, but just the JPEGs you have created?

Reply:
Correct! There is an overwriting occurring here but it is for the JPGs